picard/script/functions.py

```diff
@@ -1493,3 +1495,33 @@ def func_dateformat(parser, date_to_parse, date_format=None, date_order='ymd'):
 def func_is_multi(parser, multi):
     multi_value = MultiValue(parser, multi, MULTI_VALUED_JOINER)
     return '' if len(multi_value) < 2 else '1'
+
+
+@script_function(documentation=N_(
+    """`$parsedate(date,first_element=None)`
+
+Parses the input date string and returns the resulting date in the form "YYYY-MM-DD". The
+function will try to automatically determine the date based on the content of each element
+(e.g. text month, day greater than 12, four-digit year, etc.). When date information is
+ambiguous (e.g. 06-07-08) the system will parse it in the order "mdy", but you can force
+it to use "dmy" or "ymd" by setting `first_element` to "d" or "y" respectively.
```

Comment:
It would be more consistent with other date-related functions to use `date_order="ymd"` as default, and accept "dmy" and "mdy".

Reply:
I agree that would be more consistent with the other date-related functions.

Reply:
Even though we're putting this on hold for now, I pushed the changes to use "ymd" and "dmy" as the argument options so I didn't forget about it.